src/include/daos_obj.h

```diff
@@ -355,6 +355,54 @@ daos_obj_generate_id(daos_obj_id_t *oid, daos_ofeat_t ofeats,
 	oid->hi |= hdr;
 }
 
+typedef enum {
+	DAOS_NO_RDD,
+	DAOS_ANY_RDD,
+	DAOS_RP_RDD,
+	DAOS_EC_RDD,
+} daos_oclass_rdd_t;
+
+typedef enum {
+	DAOS_OBJ_TINY,		/** 1 grp */
+	DAOS_OBJ_SMALL,		/** ~25% */
```

Comment:
Giving a cluster with 32K targets, it is 8K targets which is not exactly "small".
I don't even feel that 1 group is "tiny" for 16+2 EC. 
So I'm thinking:
- tiny: number of data shards <= 4, or one group
- small: number of data shards <=16
- medium: number of data shards <= 32
- large: number of data shards <= 64
- scale_regular: min(total, max(128, 25% * total))
- scale_high: min(total, max(256, 50% * total))
- scale_extreme: min(total, max(1024, 80% * total))
- scale_max: total

Reply:
An example:  for EC(2+1), "small" means 8 groups, but EC(16+2), tiny/small means 1 group.

Reply:
Probably still too many options for a middleware to choose ... but a good start.

Reply:
i can rename to regular/high/extreme
but for the other ones, i think if the developers understand what they mean, then they can use the oclass directly vs the hints